magma/display.py

```diff
@@ -0,0 +1,142 @@
+from .circuit import _definition_context_stack
+from .t import Type
+
+
+class Event:
```

Comment:
we can make all these classes (Event, Posedge, Negedge, Display, File) private, and just have the functions be public